crates/red_knot_python_semantic/src/semantic_index/predicate.rs

```diff
@@ -85,3 +87,60 @@ impl<'db> PatternPredicate<'db> {
         self.file_scope(db).to_scope_id(db, self.file(db))
     }
 }
+
+/// A "placeholder predicate" that is used to model the fact that the boundness of a
+/// (possible) definition or declaration caused by a `*` import cannot be fully determined
+/// until type-inference time. This is essentially the same as a standard visibility constraint,
+/// so we reuse the [`Predicate`] infrastructure to model it.
+///
+/// To illustrate, say we have a module `a.py` like so:
```

Comment:
This is a really minor nuisance, but I would have an easier time reading this doc comment if the modules were named `module.py` and `main.py` or similar, so I don't have to remember which one is the exporting and which one is the importing module :smile:. Sorry, it's late here...